pkgs/by-name/c-/c-periphery/package.nix

```
@@ -0,0 +1,33 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "c-periphery";
+  version = "2.4.2";
+
+  src = fetchFromGitHub {
+    owner = "vsergeev";
+    repo = "c-periphery";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-uUSXvMQcntUqD412UWkMif0wLxPhpPdnMb96Pqqh/B4=";
+  };
+
+  postPatch = ''
+    substituteInPlace src/libperiphery.pc.in \
+      --replace '=''${prefix}/' '=' \
+      --replace '=''${exec_prefix}/' '='
+  '';
+
+  nativeBuildInputs = [ cmake ];
```

Maybe we could split the output?

```suggestion
  nativeBuildInputs = [ cmake ];
  
  outputs = [ "out" "lib" ];
```

---

Does it make sense since it provides only static library `libperiphery.a`?

---

It would split the output into one with headers and one with lib. I guess most of the packages that would require the headers would need the binary as well though, so I'm not sure how needed it is?

---

Indeed, usually it's useful for dynamic libraries to reduce the size of a closure. But here we have static one.

---

The advantages of reduction are still there even in the case of static builds.
Also it provides pkgconfig files, another reason to split.

---

Well, added.

---

Maybe `dev` too?

```text
results/c-periphery/
├── include
│   └── periphery
│       ├── gpio.h
│       ├── gpio_internal.h
│       ├── i2c.h
│       ├── led.h
│       ├── mmio.h
│       ├── pwm.h
│       ├── serial.h
│       ├── spi.h
│       └── version.h
├── lib
│   └── pkgconfig
│       └── libperiphery.pc
└── nix-support
    └── propagated-build-inputs

results/c-periphery.lib/
└── lib
    └── libperiphery.a
```

---

added